backend/controller/controller.go

```diff
@@ -1002,6 +1013,53 @@ func (s *Service) reconcileRunners(ctx context.Context) (time.Duration, error) {
 	return time.Second, nil
 }
 
+func (s *Service) executeAsyncCalls(ctx context.Context) (time.Duration, error) {
+	logger := log.FromContext(ctx)
+	logger.Tracef("Acquiring async call")
+
+	call, err := s.dal.AcquireAsyncCall(ctx)
```

Comment:
What happens in the case where async calls are registered at a faster rate than `number of controllers * 2 seconds`? Is that the limit that this code supports?

Reply:
That could be the first use of returning a different delay - return 0s if we've executed a call, 2s otherwise.

Reply:
Or we could keep calling until it runs out.

Reply:
Both make sense to me 👍

Reply:
We could also just spin up more workers dynamically

Reply:
Other thing on my mind here:
Async use cases may lean towards the longer running use cases (once this all works with cron, async + callback, ...).
Do we want the controller to attempt async calls serially (1 controller = 1 async call can running), or should it switch to a more concurrent approach

Reply:
I don't understand, what's an example of the situation you're envisaging that isn't covered?

Reply:
Ah not a new thing that isn't covered, just thinking about when this limit would be most restrictive. In that the limit is a lot easier to hit if the async task takes a while to execute.
And thinking about when the callback pattern is most used, its for when the task may take a while (thinking ahead to when async task is used for callbacks).